networkx/algorithms/coloring/greedy_coloring.py

```diff
@@ -209,29 +209,44 @@ def strategy_saturation_largest_first(G, colors):
 
     """
     distinct_colors = {v: set() for v in G}
-    for i in range(len(G)):
-        # On the first time through, simply choose the node of highest degree.
-        if i == 0:
-            node = max(G, key=G.degree)
-            yield node
-            # Add the color 0 to the distinct colors set for each
-            # neighbors of that node.
-            for v in G[node]:
-                distinct_colors[v].add(0)
-        else:
-            # Compute the maximum saturation and the set of nodes that
-            # achieve that saturation.
-            saturation = {
-                v: len(c) for v, c in distinct_colors.items() if v not in colors
-            }
-            # Yield the node with the highest saturation, and break ties by
-            # degree.
-            node = max(saturation, key=lambda v: (saturation[v], G.degree(v)))
-            yield node
-            # Update the distinct color sets for the neighbors.
-            color = colors[node]
-            for v in G[node]:
-                distinct_colors[v].add(color)
+
+    # Add the node color assignments given in colors to the
+    # distinct colors set for each neighbor of that node
+    for vertex, color in colors.items():
+        for neighbor in G[vertex]:
+            distinct_colors[neighbor].add(color)
+
+    # Check that the color assignments in colors are valid
+    # i.e. no neighboring nodes have the same color
+    if len(colors) >= 2:
+        for vertex, color in colors.items():
+            if color in distinct_colors[vertex]:
+                raise nx.NetworkXError(
+                    "Neighboring vertices must have different colors"
+                )
+
+    # If 0 nodes have been colored, simply choose the node of highest degree.
+    if len(colors) == 0:
```

Comment:
I think PEP8 standards say `if len(colors) == 0:` should be written `if not colors:`

Reply:
Ah I didn't know that convention. Will update - thanks!